### PR TITLE
[SAK-40996] Moved momentjs import out of bullhorns only block

### PR DESCRIPTION
**Pull request by adrianfish**

### CODE REVIEW

**Comment by mpellicer** [comment]
Hi @adrianfish. momentjs is imported across several tools using the "includeWebjarLibrary method" from headscripts.js.

https://github.com/sakaiproject/sakai/blob/master/library/src/webapp/js/headscripts.js#L780

Looks like you're loading this library at portal level, do you think it's worth to remove all references across the tools?

You can search for "includeWebjarLibrary('momentjs')"

**Comment by adrianfish** [comment]
I'll amend this to use that method just to fix this for 19. After, we should pull them out of the tools for 20. I'd say momentjs is pretty much the framework of choice for date management, so including it in the standard head is the way to go.

**Comment by adrianfish** [comment]
Actually, you know what? There are 8 tools using the function call. I'll remove those calls. Effectively, the include in the standard head is happening by default seeing as bullhorns it enabled by default.

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm [hunk 1]
```diff
@@ -64,12 +64,12 @@
         <link href="${pageScriptPath}jquery/cluetip/1.2.10/css/jquery.cluetip.css$!{portalCDNQuery}" rel="stylesheet">
         <link href="${pageWebjarsPath}qtip2/3.0.3/jquery.qtip.min.css$!{portalCDNQuery}" rel="stylesheet">
         <link href="${pageWebjarsPath}pnotify/3.2.1/pnotify.css$!{portalCDNQuery}" rel="stylesheet">
-	<link href="${pageWebjarsPath}pnotify/3.2.1/pnotify.buttons.css$!{portalCDNQuery}" rel="stylesheet">
+        <link href="${pageWebjarsPath}pnotify/3.2.1/pnotify.buttons.css$!{portalCDNQuery}" rel="stylesheet">
         <link href="${pageWebjarsPath}cropper/4.0.0/dist/cropper.min.css$!{portalCDNQuery}" rel="stylesheet">
         <script src="${pageSkinRepo}/${pageSkin}/js/lib/modernizr.js$!{portalCDNQuery}"></script>
+        <script src="${pageWebjarsPath}momentjs/2.22.2/min/moment-with-locales.min.js$!{portalCDNQuery}"></script>
```
Comment (ern) [inline portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm]: could we update this to use includeWebjarLibrary as the best thing here is that we don't hardcode the versions other than headscript.js

Comment (adrianfish) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm]: Yep, can do.

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm [hunk 2]
```diff
@@ -64,12 +64,13 @@
         <link href="${pageScriptPath}jquery/cluetip/1.2.10/css/jquery.cluetip.css$!{portalCDNQuery}" rel="stylesheet">
         <link href="${pageWebjarsPath}qtip2/3.0.3/jquery.qtip.min.css$!{portalCDNQuery}" rel="stylesheet">
         <link href="${pageWebjarsPath}pnotify/3.2.1/pnotify.css$!{portalCDNQuery}" rel="stylesheet">
-	<link href="${pageWebjarsPath}pnotify/3.2.1/pnotify.buttons.css$!{portalCDNQuery}" rel="stylesheet">
+        <link href="${pageWebjarsPath}pnotify/3.2.1/pnotify.buttons.css$!{portalCDNQuery}" rel="stylesheet">
         <link href="${pageWebjarsPath}cropper/4.0.0/dist/cropper.min.css$!{portalCDNQuery}" rel="stylesheet">
         <script src="${pageSkinRepo}/${pageSkin}/js/lib/modernizr.js$!{portalCDNQuery}"></script>
+        <script type="text/javascript">includeWebjarLibrary('momentjs');</script>
+        <script src="${pageWebjarsPath}momentjs/2.22.2/min/moment-with-locales.min.js$!{portalCDNQuery}"></script>
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm]: Does that bring in the core library? Isn't it just the locales? If it's the whole thing, plus the locales, then the import of the core library should be removed. I'll test it later.

[the remark this answers is not in the document]
Comment (mpellicer) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm]: includeWebjarLibrary already includes "momentjs/2.22.2/min/moment-with-locales.min.js", I think we can remove the reference.

Let me test it!

[the remark this answers is not in the document]
Comment (adrianfish) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm]: Okay, cool. Thanks!

[the remark this answers is not in the document]
Comment (mpellicer) [reply portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/includeStandardHead.vm]: Is not working for me :( "headscript.js" should be included too.

image

**Comment by adrianfish** [comment]
@mpellicer @ern 

Backed off from the using includeWebjarLibrary as I can't rely on headscripts being there in all cases. The other script includes all use the path, so it's consistent at least. Down the line we'll be moving all this to ES6 modules, anyway. The include WebjarLibrary will be redundant at that point.

**Comment by adrianfish** [comment]
@mpellicer Cheers!

**Pull request merged**